contracts/wrapped_azero/lib.rs

```diff
@@ -133,23 +128,21 @@ pub mod wrapped_azero {
                 .own_code_hash()
                 .expect("Called new on a contract with no code hash");
 
-            let role_check = <Self as AccessControlled>::check_role(
-                AccountId::from(ACCESS_CONTROL_PUBKEY),
-                caller,
-                Role::Initializer(code_hash),
-                Self::cross_contract_call_error_handler,
-                Self::access_control_error_handler,
-            );
-
-            match role_check {
-                Ok(_) => ink_lang::codegen::initialize_contract(|instance: &mut WrappedAzero| {
-                    instance.metadata.name = Some("wAzero".into());
-                    instance.metadata.symbol = Some("wA0".into());
-                    instance.metadata.decimals = 12; // same as AZERO
-
-                    instance.access_control = AccountId::from(ACCESS_CONTROL_PUBKEY);
-                }),
-                Err(why) => panic!("Could not initialize the contract {:?}", why),
+            let access_control = AccountId::from(ACCESS_CONTROL_PUBKEY);
+            let access_control = AccessControlRef::from_account_id(access_control);
+            if access_control.has_role(caller, Role::Initializer(code_hash)) {
```

Comment:
[nit] I think it's already possible to return `Result<Self>` from the constructor as well so here, we could do:
```rust
self.check_role(caller, Role::Initializer(code_hash))?;
let mut metadata = metadata::Data::default();
metadata.name = Some("wAzero".into());
metadata.symbol = Some("wA0".into());
metadata.decimals = 12; // same as AZERO

Self {
    psp22: psp22::Data::default(),
    metadata,
    access_control,
}
```